app/javascript/src/components/Dashboard/Notes/NewNoteForm.jsx

```diff
@@ -20,17 +51,75 @@ export default function NewNoteForm({ onClose, refetch }) {
       initialValues={{
         title: "",
         description: "",
+        tag: "",
+        contact: "",
+        dueDate: null,
       }}
       onSubmit={handleSubmit}
-      validationSchema={yup.object({
+      validationSchema={yup.object().shape({
         title: yup.string().required("Title is required"),
         description: yup.string().required("Description is required"),
+        tag: yup.object().required("Tag is required"),
+        contact: yup.object().required("Contact is required"),
       })}
     >
-      {({ isSubmitting }) => (
+      {({ isSubmitting, setFieldValue }) => (
         <Form>
-          <Input label="Title" name="title" className="mb-6" />
-          <Textarea label="Description" name="description" rows={8} />
+          <Input label="Note Title" name="title" className="mb-6" required />
+          <Select
+            label="Tag"
+            placeholder="Select tag"
+            name="tag"
+            className="mb-6"
+            options={tagOptions}
+            required
+          />
+          <Textarea
+            className="mb-6"
+            label="Note Description"
+            name="description"
+            rows={4}
+            required
+          />
+          <Select
+            label="Assigned Contact"
+            placeholder="Select contact"
+            name="contact"
+            className="mb-6"
+            options={contactOptions}
+            required
+          />
+          <div className="flex justify-between mb-6">
+            <Label>Add Due Date to Note</Label>
+            <Switch
+              name="duedate-switch"
+              checked={addDueDate}
+              onChange={event => {
+                setAddDueDate(event.target.checked);
+                if (!event.target.checked) {
+                  setFieldValue("dueDate", null);
+                } else {
+                  setFieldValue("dueDate", new Date());
+                }
+              }}
+            />
+          </div>
+          {addDueDate && (
+            <div>
+              <Label>Due Date</Label>
+              <Field name="dueDate">
+                {() => (
+                  <DateInput
+                    format="DD-MM-YYYY"
+                    defaultValue={new Date()}
+                    onChange={value => setFieldValue("dueDate", value)}
+                    placeholder="Select due date"
+                    canClearSelection={false}
+                  />
+                )}
+              </Field>
+            </div>
```

Comment:
Instead of providing individual `mb-6` classes to block, please use `space-y-6` on the parent wrapper. 

Please refer [space between](https://tailwindcss.com/docs/space#class-reference) in tailwindCSS  for more details.